pkg/restore/split_client.go

```diff
@@ -225,19 +227,34 @@ func (c *pdClient) sendSplitRegionRequest(
 		}
 		defer conn.Close()
 		client := tikvpb.NewTikvClient(conn)
-		resp, err := client.SplitRegion(ctx, &kvrpcpb.SplitRegionRequest{
+		var resp *kvrpcpb.SplitRegionResponse
+		failpoint.Inject("not-leader-error", func(prob failpoint.Value) {
+			if rand.Int()%100 < prob.(int) {
```

Comment:
1. the probabilistic check is natively supported by failpoint (set the failpoint value to `50%return` rather than `return(50)`). search for "Supported failpoint environment variable" in https://github.com/pingcap/failpoint/
2. i don't think this should be probabilistic.

Reply:
Yes... I omitted this part of docs... Using count would be a better solution.